synapse/handlers/federation.py

```diff
@@ -1187,7 +1187,7 @@ def do_invite_join(self, target_hosts, room_id, joinee, content):
             # Check whether this room is the result of an upgrade of a room we already know
             # about. If so, migrate over user information
             predecessor = yield self.store.get_room_predecessor(room_id)
-            if not predecessor:
+            if not predecessor or "room_id" not in predecessor:
```

Comment:
there's no guarantee that `predecessor["room_id"]` is a str, so this isn't watertight, but :man_shrugging: